generate/hcl.go

```diff
@@ -44,6 +60,21 @@ func FromHCL(fs afero.Fs, path string, opt Options) (*graph.Graph, error) {
 		return nil, errors.New(diags.Error())
 	}
 
+	managedResources := make(map[string]*configs.Resource)
+	for rk, rv := range mod.ManagedResources {
+		managedResources[rk] = rv
+	}
```

Comment:
One thing we are doing right now on the State is check if we have to prefix a resource with the module because the same resource could exist on another module `aws_instance.front` could be in multiple modules.

We only do that when we have more than 1 module we should try to do the same as we'll find the same issue.